std/string.d

```diff
@@ -4722,46 +4722,76 @@ string[string] abbrev(string[] values) @safe pure
  * line is returned.
  *
  * Params:
- *    str = string to be analyzed
+ *    str = string or InputRange to be analyzed
  *    tabsize = number of columns a tab character represents
  *
  * Returns:
  *    column number
  */
 
-size_t column(S)(S str, in size_t tabsize = 8) @safe pure @nogc if (isSomeString!S)
+size_t column(Range)(Range str, in size_t tabsize = 8)
+    if (!isStaticArray!Range &&
+        (is(Range : const  char[]) ||
+         is(Range : const wchar[]) ||
+         is(Range : const dchar[]) ||
+         isInputRange!Range &&
+             (is(Unqual!(ElementEncodingType!Range) ==  char) ||
+              is(Unqual!(ElementEncodingType!Range) == wchar) ||
+              is(Unqual!(ElementEncodingType!Range) == dchar))
+        )
+       )
 {
-    import std.uni : lineSep, paraSep;
-
-    size_t column;
+    static if (is(Unqual!(ElementEncodingType!Range) == char))
+    {
+        // decoding needed for chars
+        import std.utf: byDchar;
 
-    foreach (dchar c; str)
+        return str.byDchar.column(tabsize);
+    }
+    else
     {
-        switch (c)
+        // decoding not needed for wchars and dchars
+        import std.uni : lineSep, paraSep, nelSep;
+
+        size_t column;
+
+        foreach (const c; str)
         {
-        case '\t':
-            column = (column + tabsize) / tabsize * tabsize;
-            break;
+            switch (c)
+            {
+                case '\t':
+                    column = (column + tabsize) / tabsize * tabsize;
+                    break;
 
-        case '\r':
-        case '\n':
-        case paraSep:
-        case lineSep:
-            column = 0;
-            break;
+                case '\r':
+                case '\n':
+                case paraSep:
+                case lineSep:
+                case nelSep:
+                    column = 0;
+                    break;
 
-        default:
-            column++;
-            break;
+                default:
+                    column++;
+                    break;
+            }
         }
+        return column;
     }
-    return column;
 }
 
 ///
 unittest
 {
+    import std.utf : byChar, byWchar, byDchar;
+
     assert(column("1234 ") == 5);
+    assert(column("1234 "w) == 5);
+    assert(column("1234 "d) == 5);
+
+    assert(column("1234 ".byChar()) == 5);
+    assert(column("1234 "w.byWchar()) == 5);
+    assert(column("1234 "d.byDchar()) == 5);
 
```

Comment:
They won't be tested anyway. column() only uses the most basic features of an InputRange (what `foreach` does).